cumulus/parachain-template/runtime/src/xcm_config.rs

```diff
@@ -83,11 +83,11 @@ parameter_types! {
 	pub const MaxAssetsIntoHolding: u32 = 64;
 }
 
-match_types! {
-	pub type ParentOrParentsExecutivePlurality: impl Contains<MultiLocation> = {
-		MultiLocation { parents: 1, interior: Here } |
-		MultiLocation { parents: 1, interior: X1(Plurality { id: BodyId::Executive, .. }) }
-	};
+pub struct ParentOrParentsExecutivePlurality;
+impl Contains<Location> for ParentOrParentsExecutivePlurality {
+	fn contains(location: &Location) -> bool {
+		matches!(location.unpack(), (1, []) | (1, [Plurality { id: BodyId::Executive, .. }]))
```

Comment:
Not being able to write `Here` and instead a `[]` looks pretty bad UX. Could we make a type alias for `Here`? Something like `type Here = [Junction; 0];`

Reply:
Doesn't look that bad to me, You're matching on a slice, so an empty slice is a lack of junctions.
Also, it wouldn't be a type alias that's needed, it would be a constant `const HERE: &'static [Junction] = [];`.
I can add it if using that constant seems better.

Reply:
Sure, but an empty array by itself doesn't have any semantic meaning other than a lack of junctions. You're then making users think and infer that lack of junctions => Here, and it surely is going to be tough for those who are upgrading from v3 to keep remembering all these different quirks about the change in the language.

Reply:
This doesn't seem to have been resolved.